src/components/Divider.tsx

```diff
@@ -1,70 +1,182 @@
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
-  const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.querySelector('[data-divider=true]')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
+  useEffect(() => {
+    /** Calculates and updates the Divider's width based on sibling thought widths. */
+    const updateDividerWidth = () => {
+      if (!dividerRef.current) return
+
+      if (isOnlyChild && !isTableView) {
+        setDividerWidth(DIVIDER_MIN_WIDTH)
+        return
+      }
+
+      let widths: number[] = []
+      if (isTableView && isOnlyChild) {
+        if (thoughtsAtSameDepth.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(thoughtsAtSameDepth)
+      } else {
+        // Non-Table View or Divider is not the only item
+        const siblingThoughts = children.filter(child => child.id !== dividerId && !isDivider(child.value))
+
+        if (siblingThoughts.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(siblingThoughts)
+      }
+
+      setDividerWidth(Math.round(Math.max(...widths, DIVIDER_MIN_WIDTH)))
+    }
+
+    updateDividerWidth() // Initial call to set the width
+
+    /** Attach resize observers to relevant thought elements. */
+    const observeResize = () => {
+      // Clear any existing observers
+      resizeObservers.current.forEach(observer => observer.disconnect())
+      resizeObservers.current = []
+
+      let thoughtsToObserve: { id: ThoughtId }[] = []
+
+      if (isTableView && isOnlyChild) {
+        thoughtsToObserve = thoughtsAtSameDepth
+      } else {
+        thoughtsToObserve = children.filter(child => child.id !== dividerId && !isDivider(child.value))
+      }
+
+      thoughtsToObserve.forEach(thought => {
+        const thoughtElement = document.querySelector(
+          `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+        )
+        if (thoughtElement) {
+          const resizeObserver = new ResizeObserver(() => {
+            updateDividerWidth()
+          })
+          resizeObserver.observe(thoughtElement)
+          resizeObservers.current.push(resizeObserver)
+        }
       })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
     }
-  }
 
-  useEffect(setStyle)
+    observeResize()
```

Comment:
Is there a reason for creating an `observeResize` function and then immediately calling it only once? I think it would make sense to simplify this by removing the `observeResize` function declaration and call, and unindenting the contents of the function by one.